src/NServiceBus.Core/Pipeline/Pipeline.cs

```diff
@@ -42,6 +42,7 @@ public Pipeline(IServiceProvider builder, PipelineModifications pipelineModifica
 
         public Task Invoke(TContext context)
         {
+            context.Extensions.Behaviors = behaviors;
```

Comment:
Should I add a comment here that explains this monster?

Reply:
Definitely.

Reply:
Done